docs/articles/deployment/container.md

```diff
@@ -1,25 +1,164 @@
-It is very simple to run a Bytewax dataflow program in a container. In this section we will describe how to do that.
+In this chapter we are going to build a Docker image where we can run a dataflow, and present the prebuilt images we offer.
+
+## Your dataflow
+Let's start with a simple dataflow.
+
+Create a new file `dataflow.py` with the following content:
+
+```python
+from bytewax import operators as op
+from bytewax.connectors.stdio import StdOutSink
+from bytewax.dataflow import Dataflow
+from bytewax.testing import TestingSource
+
+flow = Dataflow("test")
+inp = op.input("in", flow, TestingSource(list(range(5))))
+op.output("out", inp, StdOutSink())
+```
+
+Run it locally to check everything's working fine:
+```
+$ python -m bytewax.run dataflow
+0
+1
+2
+3
+4
+```
+
+## Docker image
+You can run the dataflow inside a Docker image.
+You'll need an image with python support, and can just install bytewax and run the dataflow.
+
+Create a `Dockerfile` with the following content:
+```Dockerfile
+# Start from a debian slim with python support
+FROM python:3.11-slim-bullseye
+# Setup a workdir where we can put our dataflow
+WORKDIR /bytewax
+# Install bytewax and the dependencies you need here
+RUN pip install bytewax
+# Copy the dataflow in the workdir
+COPY dataflow.py dataflow.py
+# And run it
+# Note: use `-u` too to let python run unbuffered,
+# which causes a number of issues on Docker.
+CMD ["python", "-um", "bytewax.run", "dataflow"]
+```
+
+Now you can build the image:
+```bash
+docker build . -t bytewax-custom
+```
+
+And check that everything's working:
+```bash
+docker run --rm bytewax-custom
+```
+
+## Example: docker compose, kafka connector and redpanda
+
+Modify the Dockerfile to install optional `kafka` dependencies in bytewax:
+```diff
+- RUN pip install bytewax
++ RUN pip install bytewax[kafka]
+```
+
+Rebuild the image with:
+
+```bash
+docker build . -t bytewax-custom
+```
+
+Modify the dataflow to read data from a kafka topic rather than the testing input:
+
+```python
+from bytewax import operators as op
+from bytewax.connectors.kafka import operators as kop
+from bytewax.connectors.stdio import StdOutSink
+from bytewax.dataflow import Dataflow
+
+flow = Dataflow("test")
+inp = kop.input("in", flow, brokers=["redpanda:9092"], topics=["in_topic"])
+op.output("out", inp, StdOutSink())
+```
+
+Now we can create a docker-compose file that runs both a `Redpanda` instance and our dataflow.
```

Comment:
Let's add a note here that we are basically using the same standard redpanda docker compose almost exactly

Reply:
ok, should I also link redpanda's docs here, or just mention them?